src/gui/qgsexpressionbuilderwidget.cpp

```diff
@@ -440,6 +443,26 @@ void QgsExpressionBuilderWidget::loadRecent( const QString &collection )
   }
 }
 
+void QgsExpressionBuilderWidget::loadLayers()
+{
+  QMap<QString, QgsMapLayer *> layers = QgsProject::instance()->mapLayers();
```

Comment:
Ok done, but I've left it to default to QgsProject::instance(). This widget (and the accompanying dialog) is scattered all through gui/app. It's not a trivial task to catch all those uses and flip them to manually setting the project)